src/main/java/org/generationcp/middleware/service/impl/inventory/TransactionServiceImpl.java

```diff
@@ -200,11 +208,28 @@ public void depositLots(final Integer userId, final Set<Integer> lotIds, final L
 			} else {
 				transaction.setCommitmentDate(0);
 			}
-			daoFactory.getTransactionDAO().save(transaction);
+			this.daoFactory.getTransactionDAO().save(transaction);
+
+			if (lotDepositRequestDto.getSourceStudy() != null) {
+				// Create experiment transaction records when lot and deposit are created in the context of study.
+				this.createExperimentTransaction(lotDto.getGid(), germplasmStudySourceMap, transaction);
+			}
+
 		}
 
 	}
 
+	private void createExperimentTransaction(final Integer gid, final Map<Integer, GermplasmStudySource> germplasmStudySourceMap,
+		final Transaction transaction) {
```

Comment:
Probably offer `ExperimentTransactionType` as a parameter so as not to hardcode to `HARVESTING` type

Reply:
Done